pkg/ebpf/generic_map_test.go

```diff
@@ -0,0 +1,526 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package ebpf
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cilium/ebpf"
+	"github.com/stretchr/testify/require"
+
+	ebpfkernel "github.com/DataDog/datadog-agent/pkg/security/ebpf/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+)
+
+func TestBatchAPISupported(t *testing.T) {
+	// Batch API is supported on kernels >= 5.6, so make sure that in those cases
+	// it returns true
+	kernelVersion, err := ebpfkernel.NewKernelVersion()
+	require.NoError(t, err)
+
+	if kernelVersion.IsRH7Kernel() || kernelVersion.IsRH8Kernel() {
+		// Some of those kernels have backported the batch API, I don't want
+		// to include those specifics in unit tests that are only meant to ensure
+		// that the checks are correct in at least the basic cases.
+		t.Skip("Unknown support for batch API on RHEL kernels")
+	}
+
+	require.Equal(t, kernelVersion.Code >= ebpfkernel.Kernel5_6, BatchAPISupported())
+}
+
+func TestSingleItemIter(t *testing.T) {
+	m, err := NewGenericMap[uint32, uint32](&ebpf.MapSpec{
+		Type:       ebpf.Hash,
+		MaxEntries: 10,
+	})
+	require.NoError(t, err)
+
+	numsToPut := []uint32{1, 2, 3, 4, 5}
+	for _, num := range numsToPut {
+		require.NoError(t, m.Put(&num, &num))
+	}
+
+	var k uint32
+	var v uint32
+	numElements := 0
+	foundElements := make(map[uint32]bool)
+
+	it := m.IterateWithOptions(IteratorOptions{ForceSingleItem: true})
+	require.NotNil(t, it)
+	require.IsType(t, &genericMapItemIterator[uint32, uint32]{}, it)
+	for it.Next(&k, &v) {
+		numElements++
+		foundElements[k] = true
+	}
+
+	require.Equal(t, len(numsToPut), numElements)
+	for _, num := range numsToPut {
+		require.True(t, foundElements[num])
+	}
+}
+
+func TestBatchIter(t *testing.T) {
+	if !BatchAPISupported() {
+		t.Skip("Batch API not supported")
+	}
+
+	m, err := NewGenericMap[uint32, uint32](&ebpf.MapSpec{
+		Type:       ebpf.Hash,
+		MaxEntries: 100,
+	})
+	require.NoError(t, err)
+
+	numsToPut := uint32(50)
+	for i := uint32(0); i < numsToPut; i++ {
+		require.NoError(t, m.Put(&i, &i))
+	}
+
+	var k uint32
+	var v uint32
+	numElements := uint32(0)
+	foundElements := make(map[uint32]bool)
+
+	it := m.IterateWithOptions(IteratorOptions{BatchSize: 10})
+	require.NotNil(t, it)
+	require.IsType(t, &genericMapBatchIterator[uint32, uint32]{}, it)
+	for it.Next(&k, &v) {
+		numElements++
+		foundElements[k] = true
+	}
+
+	require.Equal(t, numsToPut, numElements)
+	for i := uint32(0); i < numsToPut; i++ {
+		require.True(t, foundElements[i])
+	}
```

Comment:
I think you could simplify the expected value checks by creating a slice of the `Put` values, creating another slice with the results of the iteration, and then doing `require.Equal` between them.